src/App.tsx

```diff
@@ -15,36 +16,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortField {
+  alphabet = 'alphabet',
+  length = 'length',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortingType: SortField | null,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortingType) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortingType) {
+        case SortField.alphabet:
+          return good1.localeCompare(good2);
+        case SortField.length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortingType, setSortingType] = useState<SortField | null>(null);
+  const [reversedState, setReversedState] = useState(false);
+  let visibleGoods = getPreparedGoods(goodsFromServer, sortingType);
+  const resetStates = () => {
+    setSortingType(null);
+    setReversedState(false);
+  };
+
+  if (reversedState) {
+    visibleGoods = visibleGoods.reverse();
+  }
```

Comment:
[CODE QUALITY] - Direct mutation of the `visibleGoods` array is not recommended. It would be better to use the `reverse` method in the same line where you define `visibleGoods`.